ConsoleProject/src/view/Frontend.cs

```diff
@@ -46,18 +49,23 @@ public class Frontend {
             };
 
             form.DragDrop += (s, e) => {
-                string[] fileList = (string[])e.Data.GetData(DataFormats.FileDrop, false);
-                if (fileList.Length > 0) { // is this IF required?
-                    try {
-                        images = new TextureConverter().ParseFile(fileList[0]); // TODO only reads first file
-                        this.LoadImages();
-                    } catch (InvalidDataException ex) {
-                        Console.WriteLine(ex);
-                        MessageBox.Show("No images found!\nIncorrect file?");
-                    }
+                string[] filesDropped = (string[])e.Data.GetData(DataFormats.FileDrop, false);
+                List<Texture> listDropped = new TextureConverter().ParseNewFile(filesDropped);
+                if (listDropped != null) {
```

Comment:
Replace null check with exception handling. Also extract the common part to a private function here. This will solve your code duplication concerns too.